gdx/src/com/badlogic/gdx/graphics/g2d/Animation.java

```diff
@@ -94,10 +95,12 @@ public T getKeyFrame (float stateTime, boolean looping) {
 			else
 				playMode = PlayMode.LOOP_REVERSED;
 		} else if (!looping && !(playMode == PlayMode.NORMAL || playMode == PlayMode.REVERSED)) {
-			if (playMode == PlayMode.LOOP_REVERSED)
+			if (playMode == PlayMode.LOOP_REVERSED) {
 				playMode = PlayMode.REVERSED;
-			else
-				playMode = PlayMode.LOOP;
+			} else if (playMode == PlayMode.LOOP) {
+				playMode = PlayMode.NORMAL;
+			}
+			// non-loop param takes no effect on LOOP_PINGPONG and LOOP_RANDOM mode
```

Comment:
This comment isn't strictly necessary, but I think that if you want this here at all, it should be `if the loop param is false, LOOP_PINGPONG and LOOP_RANDOM playModes won't change` . Again, a very minor nitpick, especially since users usually won't be reading this.